src/value/ser.rs

```diff
@@ -9,8 +9,17 @@ impl Serialize for Value {
         match *self {
             Value::Null => serializer.serialize_unit(),
             Value::Bool(b) => serializer.serialize_bool(b),
-            Value::I64(i) => serializer.serialize_i64(i),
-            Value::F64(f) => serializer.serialize_f64(f),
+            Value::Number(ref n) => {
```

Comment:
This should be implemented as `n.serialize(serializer)` to avoid the unreachable case.